src/pose_estimating.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * @file pose.h
+ * @brief Interface for representation of user's pose
+
+ * @copyright Copyright (C) 2021  Conor Begley
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SRC_POSE_ESTIMATING_H_
+#define SRC_POSE_ESTIMATING_H_
+
+#include "post_processor.h"
+
+namespace PoseEstimating {
+
+/**
+ * @brief An enum for each important pose body part
+ */
+enum Joint {
+    Head,
+    Neck,
+    Shoulder,
+    Hip,
+    Knee,
+    Foot
+}
+
+/*
+ * @brief A representation of a body part
+ *
+ * Each body part is represented as a `Pose::ConnectedJoint` to record the joints relative position and the connected joints to this joint
+ * 
+ */
+struct ConnectedJoint {
+    Joint joint;
+    /* @brief relative x co-ordinate of joint */
+    float x;
+    /* @brief relative y co-ordinate of joint */
+    float y;
+    ConnectedJoint * upper_connected_joint;
+    /* 
+     * @brief Angle(in degrees) of upper joint measured anticlockwise from bottom left corner  
+     */
```

Comment:
I'm not sure I understand this. Would you mind explaining what the angle represents? Intuitively (for me) the angle would be measured clockwise from "north":

```
  ^ North
  |
  x   <------- shoulder
   \ 
    \ 
     \
      \ 
        x   <------- elbow
```
In this example the elbow is 140 degrees from the shoulder -- basically just a bearing

Reply:
So I was basing it off the usual standard x y origin where the angles are often measured anti clockwise from "East" i.e 0-90 positive x and positive y, 90-180 positive x and negative y etc. But may just be how I was thought maths. Happy to change to whatever.

Reply:
@Ashwin-MJ @Andrew-Ritchie, do either of you have a preference for this? 

Either way it would be good to document this angle more explicitly.

Reply:
Personally I'd be inclined to agree with Miklas and use "North" as 0⁰ and go clockwise from there. As long as we document properly our decision it should be fine. 

We should also consider that users might use this from the other side, in which case everything would effectively be mirrored. Not sure if that matters right now, but worth keeping in mind

Reply:
Will mirroring be an issue? We can decide to use left or right shoulder/leg etc, but the model we use can remain the same?